osf/models/files.py

```diff
@@ -772,6 +805,46 @@ def location_hash(self):
     def archive(self):
         return self.metadata.get('archive')
 
+    def _purge(self, client=None, save=True):
+        if not client:
+            logger.warn(f'No GCS Client detected. Not purging FV {self.id}')
+            return 0
+        if self.basefilenode_set.filter(deleted__isnull=True).exists():
+            logger.warn(f'Live file detected. Not purging FV {self.id}')
+            return 0
+        if not self.location or not self.location.get('object'):
+            logger.warn(f'No valid location detected. Not purging FV {self.id}')
+            return 0
+        dup = FileVersion.objects.exclude(id=self.id).filter(location__object=self.location['object'], basefilenode__deleted__isnull=True).first()
```

Comment:
without an index on `location` this feels worrisomely slow (guessing it'll seq scan through the whole table if no matches) -- not the biggest concern for a periodic script, but we have a lot of fileversions

Reply:
An earlier iteration did not perform this query, and my time for a dry run of 1k was ~16 minutes.
The time for a 1k run with this query was ~22 minutes, a ~37% increase.

I agree it's not ideal, but the disk cost (and creation locking cost) of an index is likely worse than the incurred inefficiency. If it was an increase of an order of magnitude or more, it might be a major concern.